iphone/Classes/TiUIiPhoneProxy.m

```diff
@@ -226,10 +226,12 @@ -(NSString*)apiName
 
 BEGIN_UI_THREAD_PROTECTED_VALUE(statusBarHidden,NSNumber)
 result = [NSNumber numberWithBool:[[UIApplication sharedApplication] isStatusBarHidden]];
+DEPRECATED_REPLACED_REMOVED(@"UI.iPhone.statusBarHidden",@"3.1.3", @"6.0.0", @"UI.Window.fullscreen");
```

Comment:
I don't understand what you mean for the first part ? Also the Macro above it is a method. So it only gets ran if the property is set in JS.